osu.Game.Rulesets.Taiko/Difficulty/TaikoDifficultyCalculator.cs

```diff
@@ -69,17 +67,21 @@ protected override DifficultyAttributes CreateDifficultyAttributes(IBeatmap beat
 
             var colour = (Colour)skills[0];
             var rhythm = (Rhythm)skills[1];
-            var staminaRight = (Stamina)skills[2];
-            var staminaLeft = (Stamina)skills[3];
+            var stamina = (Stamina)skills[2];
 
             double colourRating = colour.DifficultyValue() * colour_skill_multiplier;
             double rhythmRating = rhythm.DifficultyValue() * rhythm_skill_multiplier;
-            double staminaRating = (staminaRight.DifficultyValue() + staminaLeft.DifficultyValue()) * stamina_skill_multiplier;
+            double staminaRating = stamina.DifficultyValue() * stamina_skill_multiplier;
 
             double staminaPenalty = simpleColourPenalty(staminaRating, colourRating);
             staminaRating *= staminaPenalty;
 
-            double combinedRating = locallyCombinedDifficulty(colour, rhythm, staminaRight, staminaLeft, staminaPenalty);
+            if (beatmap.BeatmapInfo.Ruleset.OnlineID == 0 && colourRating < 0.05)
+            {
+                staminaPenalty *= 0.25;
+            }
```

Comment:
Two questions here:

* Why are converts getting directly targeted like this only in regards to one skill? I believe I've heard things that taiko players don't really value converts, but I'd still like to believe that a map is a map is a map, and it shouldn't matter all that much if it's a convert or not if there's a good enough algorithm for all maps. The commit message mentions "low colour variance" - so is this just some sort of holdover until the colour skill is redone, to ensure that values don't look bad?
* After these lines, `staminaPenalty` is only ever again used once, in `locallyCombinedDifficulty()`, while `separatedRating` uses `staminaRating` which was multiplied by `staminaPenalty` _before_ it had the convert nerf applied to it. Why is that the case?

Reply:
This is a holdover until the colour skill is done, and do fix the issues with monos (single colour) streams. In terms of its placement, it's more of a blanket fix and was utilised in total calculations rather than seperate to be easily controlled. I can fix that up if you'd like however for code quality

Reply:
I don't mind the placement if it is indeed temporary, at most you can leave a TODO that this will be addressed later.

Reply:
I'll add a todo, on an other note i'll resolve the code quality primary too as soon as I can.

Reply:
todo added, resolving for now